mlir/lib/Conversion/TosaToRock/TosaToRockPass.cpp

```diff
@@ -53,7 +53,7 @@ struct TosaToRock : public impl::TosaToRockPassBase<TosaToRock> {
     target.addLegalDialect<rock::RockDialect, tosa::TosaDialect,
                            tensor::TensorDialect,
                            bufferization::BufferizationDialect>();
-    target.addIllegalOp<tosa::Conv2DOp, tosa::MatMulOp>();
+    target.addIllegalOp<tosa::Conv2DOp, tosa::MatMulOp, tosa::ReduceSumOp>();
```

Comment:
Dynamically legal, perhaps? We don't want to actually fail on non-sum reductions?

Reply:
Not sure I follow, non-sum reductions are not affected ?

Reply:
If an op is illegal, and you fail to rewrite it, that's a pass failure.

Reply:
yea and the illegal op specifically is tosa::ReduceSumOp.. so what is the relevance to non-sum reductions ?

Reply:
... my not reading correctly and thinking that you were working on a general `tosa.reduce`